components/CategoriesList/CategoriesList.tsx

```diff
@@ -52,6 +52,14 @@ function CategoriesList({ categories, showAllCategories = false, isStatic, linkC
         ];
     }, [categories, showExtraCategories, currentLocale]);
 
+    useEffect(() => {
+        if (showAllCategories) {
+            setShowExtraCategories(true);
+        } else {
+            setShowExtraCategories(false);
+        }
+    }, [showAllCategories]);
```

Comment:
❌ 
You don't need an effect here, you can just derive state from props.
E.g. on line 24 you can update the condition to `showExtraCategories || showAllCategories`

Reply:
Ah, now I see. Then please add a comment explaining the need to use this effect, so that it doesn't get "refactored" by acccident in the future :)